deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.kt

```diff
@@ -29,17 +31,61 @@ class DeepLinkEntry(val uriTemplate: String,
 
     private val parameterMap: MutableMap<DeepLinkUri, Map<String, String>> = mutableMapOf()
 
+    private val firstConfigurablePathSegmentIndex: Int by lazy { uriTemplate.indexOf(configurablePathSegmentPrefixChar) }
+    private val firstPlaceholderIndex: Int by lazy { uriTemplate.indexOf(componentParamPrefixChar) }
+    private val firstNonConcreteIndex: Int by lazy {
+        if (firstPlaceholderIndex == -1 && firstConfigurablePathSegmentIndex == -1) {
+            -1
+        } else {
+            if (firstConfigurablePathSegmentIndex == -1) {
+                firstPlaceholderIndex
+            } else {
+                if(firstPlaceholderIndex == -1){
+                    firstConfigurablePathSegmentIndex
+                } else {
+                    min(firstConfigurablePathSegmentIndex, firstPlaceholderIndex)
+                }
+            }
+        }
+    }
+
     /**
      * Generates a map of parameters and the values from the given deep link.
      *
      * @param inputUri the intent Uri used to launch the Activity
      * @return the map of parameter values, where all values will be strings.
      */
     fun getParameters(inputUri: DeepLinkUri): Map<String, String> {
-        return parameterMap.get(inputUri) ?: emptyMap()
+        return parameterMap[inputUri] ?: emptyMap()
     }
 
     fun setParameters(deepLinkUri: DeepLinkUri, parameterMap: Map<String, String>) {
         this.parameterMap[deepLinkUri] = parameterMap
     }
+
+    override fun toString(): String {
+        return "uriTemplate: $uriTemplate type: $type activity: ${activityClass.simpleName} method: $method parameters: $parameterMap"
+    }
+
+    /**
+     * Whatever template has the first placeholder (and then configurable path segment) is the less
+     * concrete one.
+     * Because if they would have been all in the same index those elements would have been on the
+     * same level and in the same "list" of elements we compare in order.
+     * In this case the one with the more concete element would have won and the same is true here.
+     */
+    fun moreConcreteThan(compare: DeepLinkEntry): Int {
```

Comment:
implement `Comparable` instead?

Reply:
Yeah, I had that, and then I refactored it way. :-) I agree.